Framework/Kernel/src/Matrix.cpp

```diff
@@ -1613,7 +1613,7 @@ void fillFromStream(std::istream &is, Kernel::Matrix<T> &in,
 
 // Symbol definitions for common types
 template class MANTID_KERNEL_DLL Matrix<double>;
-template class MANTID_KERNEL_DLL Matrix<int>;
+template class Matrix<int>;
```

Comment:
Having the ``MANTID_KERNEL_DLL` creates the warning: 
`warning: type attributes ignored after type is already defined [-Wattributes]
 template class MANTID_KERNEL_DLL Matrix<int>;`

I had similar warnings at other locations but could avoid them by rearranging the code. The reason for this seems to be the use of Matrix<int> in a template specialization: https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/src/Matrix.cpp#L812

There are other parts of the code, where this had to be done:. `TimeSeriesProperty`, `NDArrayToVector` , `NDArrayTypeIndex`